src/sentry/eventstream/kafka/backend.py

```diff
@@ -137,18 +144,34 @@ def _send(
     def requires_post_process_forwarder(self):
         return True
 
+    def reset_batch(self) -> None:
+        self._pending_futures.clear()
+        self._batch_create_time_ms = int(time.time() * 1000)
+
+    @staticmethod
+    def record_metrics(partition: int, task_kwargs: Mapping[str, Any]) -> None:
+        event_type = "transactions" if task_kwargs["group_id"] is None else "errors"
+        metrics.incr(
+            "eventstream.messages",
+            tags={"partition": partition, "type": event_type},
+        )
+
     def run_post_process_forwarder(
         self,
         consumer_group,
         commit_log_topic,
         synchronize_commit_group,
         commit_batch_size=100,
+        commit_batch_timeout_ms=1000,
```

Comment:
Do you know how long are our batches taking today? 
Asking in case setting a timeout in milliseconds could cause us to to commit more often than we were doing before.

Reply:
I don't have the exact metric which can tell me how long the batches are taking. But looking at the message processing rate per partition, around 15 out of 64 partitions are processing less than 100 messages per second which is the commit batch size right now. So for most partitions it would be the commit batch size limit being reached which would cause batches to close.